content/pytorch/concepts/tensors/tensors.md

```diff
@@ -0,0 +1,229 @@
+
+#Convert the float16 tensor to an int16 tensor using .to()
+myExFloat16Tensor = myExFloat16Tensor.to(torch.int16)
+
+print(myBoolTensor,"\n")
+print(myExFloat16Tensor)
+```
+
+Output:
+
+```shell
+tensor([[ True, False],
+        [False, False]]) 
+
+tensor([[3, 0],
+        [2, 1]], dtype=torch.int16)
+```
+
+Notice that when printing a tensor with a numerical datatype other than the default `torch.float32`, the datatype will be specified in the output.
+
+## Tensor Shapes
+
+The shape of a tensor is determined by its number of dimensions and the size of each dimension. To illustrate, the tensor `torch.rand(2, 6, 4)` has the same shape as the tensor `torch.ones(2, 6, 4)`, whereas these have different shapes from the tensors `torch.rand(2, 6)` and `torch.ones(3, 6, 5)`.
+
+Tensor shapes can be accessed via the `.shape` method:
+
+```python
+import torch
+
+myTensor = torch.zeros(2, 4, 6, 7)
+print(myTensor.shape)
+```
+
+Output:
+
+```shell
+torch.Size([2, 4, 6, 7])
+```
+
+Operations with multiple tensors, like adding or multiplying tensors, generally require the tensors involved to have the same shapes.
+
+### Altering Tensor Shapes
+
+PyTorch provides methods for altering the shape of a tensor, provided that the total number of elements in the tensor remains the same. The total number of elements in a tensor is equal to the product of all dimension sizes. So, for example, a tensor with shape [4, 1, 6] contains 4 x 1 x 6 = 24 elements.
+
+Since dimensions of with a size of 1 do not alter the total number of elements in a tensor, PyTorch provides methods for adding and removing dimensions of size 1. The method `.unsqueeze_(n)` adds a dimension of size 1 at the nth place, shifting displiced dimension sizes to the right. Conversely, the method `.squeeze_(n)` removes a dimension of size 1 at the nth place, shifting displaced dimension sizes to the left. Note that the numbering of dimensions begins at 0.
+
+The following demonstrates the effects of `.unsqueeze_()` and `.squeeze_()` on tensor shapes:
+
+```python
+import torch
+
+myTensor1 = torch.empty(8, 2, 8, 1)
+print(myTensor1.shape)
+
+myTensor1.unsqueeze_(1)
+print(myTensor1.shape)
+
+myTensor1.squeeze_(4)
+print(myTensor1.shape)
+```
+
+Output:
+
+```shell
+torch.Size([8, 2, 8, 1])
+torch.Size([8, 1, 2, 8, 1])
+torch.Size([8, 1, 2, 8])
+```
+
+One can also create new tensors by squeezing or unsqueezing other tensors. To do so, use the methods `.squeeze()` and `.unsqueeze()` (no underlines) respectively.
+
+Tensor shapes can also be changed arbitrarily using `.reshape()`, provided that the total number of elements remains constant:
+
+```python
+import torch
+
+myTensor = torch.empty(8, 2, 8, 1)
+print(myTensor.shape)
+
+myTensor = myTensor.reshape(128)
+print(myTensor.shape)
+
+myTensor = myTensor.reshape(64,2)
+print(myTensor.shape)
+```
+
+Output:
+
+```shell
+torch.Size([8, 2, 8, 1])
+torch.Size([128])
+torch.Size([64, 2])
+```
```

Comment:
``` suggestion
Output:

``` shell
torch.Size([8, 2, 8, 1])
torch.Size([128])
torch.Size([64, 2])
```
```
The output should be preceded by a, sentence like this `This code gives the following output.` or something similar to this sentence.